fgpyo/sam/__init__.py

```diff
@@ -394,14 +394,13 @@ class CigarElement:
         - operator (CigarOp): the operator of the element
     """
 
-    length: int = attr.ib()
-    operator: CigarOp = attr.ib()
+    length: int
+    operator: CigarOp
 
-    @length.validator
-    def _validate_length(self, attribute: Any, value: int) -> None:
+    def __post_init__(self) -> None:
```

Comment:
It was a typo, should have been `__attrs_post_init__`. Good catch, thanks! I added a test.

As for your top-level question, attrs is used extensively throughout fgpyo. The original mandate was to replace attrs usages with dataclasses, but as I mentioned in the top comment, this can't be done until we drop support for python < 3.10.